### DISCUSSION TITLE
tree-select组件出错

### DISCUSSION THREAD

**Discussion by niguangyjf** [ant-design/ant-design#33456] [Q&A]
![1640695789(1)](https://user-images.githubusercontent.com/63006083/147568062-c6d5becf-76e1-4809-9cca-d1753775689a.png)

**Comment by shaunhurryup** [comment]
写的很清楚了，key 和 value 选其一
https://ant.design/components/tree-select-cn/#TreeNode-props

**Reply by niguangyjf** [reply]
4.17版本的没有问题，升级到最新版本的会有这个警告，请问新版本是增加了什么新特性吗？

**Reply by shaunhurryup** [reply]
没用过 Antd ，不好回答。不过我在 [源码](https://github.com/react-component/tree-select/blob/v5.0.0-alpha.2/src/hooks/useDataEntities.ts#L27) 中发现这么一行
```ts
warning(
  !key || String(key) === String(val),
  `\`key\` or \`value\` with TreeNode must be the same or you can remove one of them. key: ${key}, value: ${val}.`,
  );
```

也许错误在生产环境下不会发生。你也可以比较一下 4.17 和 4.18 版本下的 [rc-tree-select](https://github.com/ant-design/ant-design/blob/4.18.0/package.json#L151) 区别。
